src/naive/time/mod.rs

```diff
@@ -551,6 +556,7 @@ impl NaiveTime {
             } else {
                 frac = (i64::from(frac) + rhs.num_nanoseconds().unwrap()) as u32;
                 debug_assert!(frac < 2_000_000_000);
+                let frac = unsafe { U31::new_unchecked(frac) };
```

Comment:
I'd like some comments here that clearly document how/where the invariant is upheld in release builds.

Reply:
I assumed chrono is well enough tested, so I can assume the invariant to hold. :dagger: 

If I read the code correctly it is tested that `frac < 1_000_000_000` and `rhs.num_nanoseconds()` cannot return a value >= 1e9. Is the second assumption true? Then I this as a `// SAFETY` comment.